angrmanagement/plugins/log_human_activities/log_human_activities.py

```diff
@@ -0,0 +1,136 @@
+    HumanActivityVariableRename = None  # type: Optional[type]
+    HumanActivityFunctionRename = None  # type: Optional[type]
+
+
+# TODO: add created_by field
+TODO = "TODO"
+
+
+class LogHumanActivitiesPlugin(BasePlugin):
+    def __init__(self, *args, **kwargs):
+        if not Slacrs:
+            raise Exception("Skipping LogHumanActivities Plugin. Please install Slacrs to Initialize it.")
+        super().__init__(*args, **kwargs)
+        self.session = Slacrs().session()
+        self.project_name = None
+        self.project_md5 = None
+
+    def handle_variable_rename(self, func, offset: int, old_name: str, new_name: str, type_: str, size: int):
+        """
+        Log a user's activity of variable renaming.
+        """
+        variable_rename = HumanActivityVariableRename(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            function=func._name,
+            old_name=old_name,
+            new_name=new_name,
+            created_by=TODO,
+        )
+        self.session.add(variable_rename)
+        self.session.commit()
+        l.info("Add variable rename sesssion to slacrs")
+
+    def handle_function_rename(self, func, old_name: str, new_name: str):
+        """
+        Log a user's activity of function renaming.
+        """
+        function_rename = HumanActivityFunctionRename(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            old_name=old_name,
+            new_name=new_name,
+            created_by=TODO,
+        )
+        self.session.add(function_rename)
+        self.session.commit()
+        l.info("Add function rename sesssion to slacrs, project name %s, old_name %s, new_name %s", self.project_name, old_name, new_name)
+        # result = self.session.query(HumanActivityFunctionRename).filter(HumanActivityFunctionRename.project == self.project_name).first()
+        # l.info("Query result: old_name %s, new_name %s", result.old_name, result.new_name)
+
+    def handle_click_block(self, qblock, event):
+        block_click = HumanActivityClickBlock(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            addr=qblock.addr,
+            created_by=TODO,
+        )
+        self.session.add(block_click)
+        self.session.commit()
+        l.info("Block %x is clicked", qblock.addr)
+        return False
+
+    def handle_click_insn(self, qinsn, event):
+        insn_click = HumanActivityClickInsn(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            addr=qinsn.addr,
+            created_by=TODO,
+        )
+        self._submit_to_slacrs(insn_click)
+        l.info("Instruction %x is clicked", qinsn.addr)
+        return False
+
+    def handle_raise_view(self, view):
+        # e.g., "<class 'angrmanagement.ui.views.disassembly_view.DisassemblyView'>"
+        view_name = str(view.__class__).split('.')[-1][:-2]
```

Comment:
I think you can use `view.__class__.__name__.split('.')[-1]`.

Reply:
resolved. `view.__class__.__name__` is good.